lib/connectors/memory.js

```diff
@@ -723,6 +725,61 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replaceById = function(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    return cb(err);
+  }
+
+  // Do not modify the data object passed in arguments
+  // data = Object.create(data);
```

Comment:
Please provide a detailed report on what is failing (and how) when this line is not commented out, so that we can troubleshoot.